### DISCUSSION TITLE
Pre-configured data table filters

### DISCUSSION THREAD

**Discussion by solomatnikovvan** [dbeaver/dbeaver#35933] [Ideas]
Good afternoon, dear developers.
I have several data tables that contain all the necessary information. But in practice, very often I get requests that need to be selected quickly. For example, only the top-level values for large offices over the past year. In order to do this quickly, I go to these tables and, for example, select (`year` IN ('2023')) AND (office_level IN (1)). This is not difficult to do, but there are tables in which the number of such filters exceeds 10 pieces and it takes time. Storing SQL queries is also inconvenient. What I suggest: in the table settings at the user level, make it possible to record filters that will immediately be applied to the data table when viewing data. It will just be supplied to the Expand filter panel by default and if I do not need any of this during the work, it will not cause problems, since I will always be able to edit them.

**Comment by ask9** [comment]
Why don't you want to store a list of your filters in a text file and copy-paste them to a filter-textbox to apply the needed one when necessary?
If you don't want to use copy-paste, you may use AutoHotkey to type predefined text of your filter quickly by using some hotkey or typing some special symbols of your choice on a keyboard (autoreplacements).

The problem with adding such a saved filter in DBeaver is that the stored filters are binded to some specific queries (column names AND the same data types of these column that could be changed in a database in time and the filter will be broken on attempt to apply it).

**Comment by TB-HunBoka** [comment]
If I understand you correctly, you need this functionality on Table level, you are not joining tables.
Then you can save the filter on the table with the "Save filter setting" button.
![kép](https://github.com/user-attachments/assets/32e876b8-cf66-41f7-80ce-4a8cdc67eb5e)

If you need more than one quick query on the same table I recommend to use the SQL Tempaltes function.
![kép](https://github.com/user-attachments/assets/d8419662-f355-4699-90fb-fe3045a9dc9a)

This will do exactly what @ask9 mentioned, it will replace the Name with the Pattern.
On my worklaptop I got around 30 Tempaltes, and I don't have to keep them in mind as you can call the "Context information" assitant with ctrl+shift+space, and it will give the list of tempaltes for you.
